### ISSUE TITLE
`date_range` with `_saturating` does not go until end of month

### ISSUE THREAD

**Issue by lucazanna** [pola-rs/polars#9442]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

See example below

### Reproducible example

```python
from datetime import date

pl.date_range(date(2019, 12, 31), date(2020, 5, 1), '1mo_saturating', eager=True)
Series: 'date' [date]
[
	2019-12-31
	2020-01-31
	2020-02-29
	2020-03-29
	2020-04-29
]
```

### Expected behavior

After the month of February 2020 which has 29 days, the following months also stop at 29.

I would expect to see the last date of each month. Instead the behaviour changes for the months following February

### Installed versions

<details>

```
--------Version info---------
Polars:      0.18.3
Index type:  UInt32
Platform:    Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:      3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]

----Optional dependencies----
numpy:       1.24.3
pandas:      2.0.1
pyarrow:     12.0.1
connectorx:  <not installed>
deltalake:   <not installed>
fsspec:      <not installed>
matplotlib:  3.7.1
xlsx2csv:    <not installed>
xlsxwriter:  <not installed>
```

</details>

**Comment by MarcoGorelli** [comment]
hey @lucazanna - yeah there's not really much that can be done here, you probably want
```python
In [46]: pl.date_range(date(2019, 12, 1), date(2020, 5, 1), '1mo_saturating', eager=True).dt.month_end()
Out[46]:
shape: (6,)
Series: 'date' [date]
[
        2019-12-31
        2020-01-31
        2020-02-29
        2020-03-31
        2020-04-30
        2020-05-31
]
```

**Comment by lucazanna** [comment]
hi @MarcoGorelli , 

got it and thanks for the reply. That's indeed what I was looking for.

so I understand that the current behavior is the expected one. In this case, maybe we could add a comment on the documentation? 

If I understand it well, _saturating will depend on the prior generated row.

**Comment by MarcoGorelli** [comment]
yeah if you have a suggestion on how to clarify the docs that'd be great! fancy opening a PR?

**Comment by mcrumiller** [comment]
@lucazanna this was a fierce debate that we are definitely not returning to :) Use `dt.monthend()` or, alternatively and probably slight faster, start in the next month and subtract a day, as in:

```python
pl.date_range(date(2020, 1, 1), date(2020, 6, 1), "1mo", eager=True).dt.offset_by("-1d")
```

**Comment by MarcoGorelli** [comment]
closed by  #9451
